topics/acl.md

```diff
@@ -396,6 +413,51 @@ Example:
 In the previous section it was observed how it is possible to define commands
 ACLs based on adding/removing single commands.
 
+## Selectors
+
+Redis supports additional sets of rules that are evaluated independently of the main user permissions.
+These secondary sets of permissions are called selectors and added by wrapping a set of rules within parentheses.
+Selectors are evaluated after the main user permissions are checked and are evaluated in the order they are added to the user. 
+A command is allowed if either the root user permissions or any selectors matches a command.
+
+For example, consider a user with the ACL rules `+GET ~key1 (+SET ~key2)`.
+This user is able to execute `GET key1` and `SET key2 hello`, but not `GET key2` or `SET key1 world`.
+
+Unlike the root user permissions, selectors can not be modified after they are added.
+Selectors can be removed with the `clearselectors` keyword, which removes all of the added selectors.
+Note that `clearselectors` does not remove the root user permissions.
+
+## Key permissions
+
+Redis key patterns can also be used to define how a command is able to touch a key.
+This is achieved through rules that define key permissions.
+The key permission rules take the form of `%(<permission>)~<pattern>`.
+Permissions are defined as individual characters that map to the following key permissions:
+
+* W (Write): The data stored within the key may be updated or deleted. 
+* R (Read): User supplied data from the key is processed, copied or returned. Note that this does not include metadata such as size information (example `STRLEN`), type information (example `TYPE`) or information about whether a value exists within a collection (example `SISMEMBER`). 
+
+Permissions can be composed together by specifying multiple characters. 
+Specifying the permission as 'RW' is considered full access and is analogous to just passing in `~<pattern>`.
+
+For a concrete example, consider a user with ACL rules `+@all ~app1:* (+@readonly ~app2:*)`.
+This user has full access on `app1:*` and readonly access on `app2:*`.
+However, some commands support reading data from one key, doing some transformation, and storing it into another key.
+One such command is the `COPY` command, which copies the data from the source key into the destination key.
+The example set of ACL rules is unable to handle a request copying data from `app2:user` into `app1:user`, since neither the root permission or the selector fully matches the command.
+However, using key selectors you can define a set of ACL rules that can handle this request `+@all ~app1:* %R~app2:*`.
+The first pattern is able to match `app1:user` and the second pattern is able to match `app2:user`.
+
+Which type of permission is required for a command is documented through [key specifications](/topics/key-specs#logical-operation-flags).
```

Comment:
i think there's a way to provide a direct link, maybe there's a better way than `https://redis.io/topics/key-specs#logical-operation-flags`. @itamarhaber

Reply:
ohh, never mind. i see this is the right syntax.